File: lib_common/src/main/java/com/yyxnb/common/MainThreadUtils.java
package com.yyxnb.common;

import android.os.Handler;
import android.os.Looper;

import java.io.Serializable;

/**
 * 方便地将 Runnable post 到主线程执行
 */
public class MainThreadUtils implements Serializable {

    private static Handler HANDLER = new Handler(Looper.getMainLooper());

    public static void post(Runnable runnable) {
        if (isMainThread()) {
            runnable.run();
        } else {
            HANDLER.post(runnable);
        }
    }

    public static void postDelayed(Runnable runnable, Long delayMillis) {
        HANDLER.postDelayed(runnable, delayMillis);
    }

    public static boolean isMainThread() {
        return Looper.getMainLooper() == Looper.myLooper();
    }
}

File: lib_common/src/main/java/com/yyxnb/common/utils/MainThreadUtils.kt
package com.yyxnb.common.utils

import android.os.Handler
import android.os.Looper
import java.io.Serializable

/**
 * 方便地将 Runnable post 到主线程执行
 */
object MainThreadUtils : Serializable {

    private val HANDLER = Handler(Looper.getMainLooper())
    @JvmStatic
    fun post(runnable: Runnable) {
        if (isMainThread) {
            runnable.run()
        } else {
            HANDLER.post(runnable)
        }
    }

    fun postDelayed(runnable: Runnable, delayMillis: Long) {
        HANDLER.postDelayed(runnable, delayMillis)
    }

    val isMainThread: Boolean
        get() = Looper.getMainLooper() == Looper.myLooper()

}